internal/internal.go

```diff
@@ -557,6 +557,18 @@ func (p *envoyExtAuthzGrpcServer) check(ctx context.Context, req interface{}) (*
 		}
 	}
 
+	if resp.DynamicMetadata == nil {
+		resp.DynamicMetadata = &_structpb.Struct{
+			Fields: map[string]*_structpb.Value{},
+		}
+	}
+
+	resp.DynamicMetadata.Fields["decision_id"] = &_structpb.Value{
+		Kind: &_structpb.Value_StringValue{
+			StringValue: result.DecisionID,
+		},
+	}
+
```

Comment:
What if we first check if the decision id is defined and if it is then we include it in the dynamic metadata? Currently we'll include dynamic metadata always and it could possibly not be that useful in some cases.

Reply:
The decision id is always a uuid4, if it errors the error is returned by the check function so it looks to me like that would be a redundant check.